pilot/pkg/xds/workloadentry.go

```diff
@@ -121,6 +130,38 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 	}, features.WorkloadEntryCleanupGracePeriod)
 }
 
+// UpdateWorkloadEntryHealth updates the associated WorkloadEntries health status
+// based on the corresponding health check performed by istio-agent.
+func (sg *InternalGen) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthEvent) {
+	if !features.WorkloadEntryHealthChecks {
+		return
+	}
+	// we assume that the workload entry exists
+	entryName := autoregisteredWorkloadEntryName(proxy)
```

Comment:
Is the manually created case going to be handled in a different PR?

Reply:
i was under the impression that all of them would be auto-registered...How do I get the name of the manually created WorkloadEntry?

Reply:
https://docs.google.com/document/d/1G4s_PrmvkwCKoH_kXVeSUAszsaj3ZfpIlraSdeTSPKU/edit#bookmark=id.t6ju2m1mphe6

Something can be added to the proxyConfig, we can use an ISTIO_META_, or we could lookup by ip. For the last option, I'm not sure what the perf is on field selectors like this; I doubt they're indexed so it could be a bit slow at scale.

Reply:
sorry, for the third option, lookup to where? Is there a DNS thing for WorkloadEntry names?

Reply:
The workload entries have IP address in the spec

Reply:
we can do auto registered ones only for now to make this simpler